{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this test run:\n",
    "\n",
    "1) Generate test files for a single sample\n",
    "    * VCF and aligned haplotagged bam.\n",
    "2) Run pileup using unaltered CpG tools v1.2.0, altered v1.2.0, and v2.3.2\n",
    "3) For v2.3.2 results, filter out CpGs based on the presence of germline variants that cause CpG decay and \"phantom CpGs\" which are generated by v2.3.2 on the haplotype with the reference allele(s). \n",
    "    * Identify which haplopype a given variant is mapped to based on it's phase, and identify if it generates a denovo CpG or destroys a reference CpG site.\n",
    "    * On a given haplotype, if the variant generates a denovo CpG site, the variant allele's CpG data should be kept, while the CpG sites in the in the corresponding reference allele should be removed. \n",
    "    * If the variant causes the loss of  a CpG site on an allele, the CpG pileup for that site should be filtered out for that haplotype. \n",
    "    * The exculded CpGs are written to a separate output file for each haplotype with their accompanying column data and summarized (plot of distribution of the methylation values, should be unimodal at 0.)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/gs/gsfs0/users/greally-lab/David/LRS_data_analysis_tutorials/CpG_pileup_QC/bash\n"
     ]
    }
   ],
   "source": [
    "# Set up environment\n",
    "import os\n",
    "import sys\n",
    "\n",
    "# Set up paths\n",
    "base_dir = \"/gs/gsfs0/users/greally-lab/David/LRS_data_analysis_tutorials/CpG_pileup_QC\"\n",
    "bash_script_path = os.path.join(base_dir, \"bash\")\n",
    "\n",
    "print(bash_script_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 8160736\n"
     ]
    }
   ],
   "source": [
    " # Generate test files for SPM161, run pileup, and run QC on CpGs.\n",
    " # test region is a small section of chr1. \n",
    "vcf_file=\"/gs/gsfs0/users/greally-lab/David/6_base-seq_SC1/WGS-analysis/outputs_compiled/20240501-results/sample_phased_small_variant_vcfs/SPM161/SPM161.GRCh38.deepvariant.phased.vcf.gz\"\n",
    "bam_file=\"/gs/gsfs0/users/greally-lab/David/6_base-seq_SC1/WGS-analysis/outputs_compiled/20240501-results/merged_haplotagged_bam/SPM161/SPM161.m84103_231128_221210_s1.hifi_reads.bc2037.GRCh38.aligned.haplotagged.bam\"\n",
    "ref_fasta=\"/gs/gsfs0/users/greally-lab/David/reference_genomes/hg38/GRCh38_full_analysis_set_plus_decoy_hla.fa\"\n",
    "output_prefix=\"SPM161\"\n",
    "output_dir=\"/gs/gsfs0/users/greally-lab/David/LRS_data_analysis_tutorials/CpG_pileup_QC/outputs/test_files/12-10-24\"\n",
    "cpg_tools_dir=\"/gs/gsfs0/users/greally-lab/David/software/pb-CpG-tools-v2.3.2-x86_64-unknown-linux-gnu\"\n",
    "\n",
    "!sbatch $bash_script_path/1_generate_test_data.sh \\\n",
    "    $vcf_file \\\n",
    "    $bam_file \\\n",
    "    $ref_fasta \\\n",
    "    $output_prefix \\\n",
    "    $output_dir \\\n",
    "    $cpg_tools_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 8160754\n"
     ]
    }
   ],
   "source": [
    "### Run QC on the full bed file for SPM161\n",
    "\n",
    "# Run QC on the CpG pileups to remove phantom CpGs.\n",
    "vcf_file=\"/gs/gsfs0/users/greally-lab/David/6_base-seq_SC1/WGS-analysis/outputs_compiled/20240501-results/sample_phased_small_variant_vcfs/SPM161/SPM161.GRCh38.deepvariant.phased.vcf.gz\"\n",
    "ref_fasta=\"/gs/gsfs0/users/greally-lab/David/reference_genomes/hg38/GRCh38_full_analysis_set_plus_decoy_hla.fa\"\n",
    "prefix=\"SPM161.GRCh38\"\n",
    "bed_dir=\"/gs/gsfs0/users/greally-lab/David/6_base-seq_SC1/WGS-analysis/outputs_compiled/20240501-results/cpg_pileup_beds/SPM161\"\n",
    "output_dir=\"/gs/gsfs0/users/greally-lab/David/LRS_data_analysis_tutorials/CpG_pileup_QC/outputs/qc_pileups_SPM161-full\" \n",
    "\n",
    "!sbatch $bash_script_path/qc_full-sample.sh \\\n",
    "    $vcf_file \\\n",
    "    $ref_fasta \\\n",
    "    $prefix \\\n",
    "    $bed_dir \\\n",
    "    $output_dir "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 8161866\n"
     ]
    }
   ],
   "source": [
    "# test segment\n",
    "input_dir=\"/gs/gsfs0/users/greally-lab/David/LRS_data_analysis_tutorials/CpG_pileup_QC/outputs/qc_pileups_SPM161-full/filtered_beds/passing/seg_test\"\n",
    "sample_name=\"SPM161.GRCh38\"\n",
    "output_dir=\"/gs/gsfs0/users/greally-lab/David/LRS_data_analysis_tutorials/CpG_pileup_QC/outputs/qc_pileups_SPM161-full/filtered_beds/passing/seg_test/seg_output\"\n",
    "\n",
    "\n",
    "!sbatch $bash_script_path/3_segment.sh $input_dir $sample_name $output_dir\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "anc_vig",
   "language": "python",
   "name": "anc_vig"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
